### DISCUSSION TITLE
Regions shown on the map are different from table and chart

### DISCUSSION THREAD

**Discussion by christoph-LSN** [open-sdg/open-sdg#1692] [Q&A]
@brockfanning @MoBosse @otis-bath

Because of an error in my GeoJSON I had to take a look in it.

What I have noticed is that the region names come out of the GeoJSON file and not from the data itself

`"type": "Feature",
      "properties": {
        "AGS": "K03101",
        "GEN": "Braunschweig"`

In this case on the map is shown "Braunschweig" in the table is shown "Braunschweig, Stadt" because in the GeoJSON GEN = Braunschweig.

I thougt the the value and the region name came out of the data cig the AGS as an identifier....

So may be it would be nice to get the region names from the data too...

[my GeoJSON](https://github.com/christoph-LSN/IM_Data/blob/develop/NI_Kurz_diss.geojson)
[Example Indicator](https://christoph-lsn.github.io/MT_Site/3-1-2/)

**Comment by brockfanning** [comment]
This is true - the maps rely on the GeoJSON and do not directly see the data source. This feature is possible though - by adding some functionality to the OutpuGeoJson class in sdg-build. It would be a parameter like "name_from_data_column", where you could specify a particular data column that should be used as the name. And then whatever name was already in the GeoJSON would be replaced with whatever is in that data column.

**Comment by christoph-LSN** [comment]
OK, but this will be specific for any user. My region names column in the data is called "Gebietseinheit" may be yours is called "region" and the chinese people have another name for the column..... so this would be quit tricky I think

**Reply by brockfanning** [reply]
I was thinking that the "name_from_data_column" parameter would actually include the name of the column. So for example the data config might look like:

```
...
map_layers:
  - geojson_file: my-geojson-file.json
    name_from_data_column: Gebietseinheit
    id_property: my_id_prop
    output_subfolder: my_subfolder
    filename_prefix: indicator_
...
```

So instead of using "name_property" you would use "name_from_data_column" which tells it to consult that particular data column for any names.

**Reply by christoph-LSN** [reply]
@brockfanning No, adding this line in my [config_data.yml](https://github.com/christoph-LSN/IM_Data/blob/develop/config_data.yml)

`name_from_data_column: Gebietseinheit`

shows this error

 File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/sdg/open_sdg.py", line 353, in open_sdg_prep
    outputs.append(sdg.outputs.OutputGeoJson(**geojson_kwargs))
TypeError: __init__() got an unexpected keyword argument 'name_from_data_column'
Error: Process completed with exit code 1.
